package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,605 @@
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+
+r"""
+Default Guesser
+================
+.. _DefaultGuesser:
+
+DefaultGuesser is a generic guesser class that have basic guessing methods.
+This class is a general purpose guesser that can be used with most topologies,
+but being generic make it the less accurate among all guessers.
+
+
+
+
+
+Classes
+-------
+
+.. autoclass:: DefaultGuesser
+   :members:
+   :inherited-members:
+
+"""
+from .base import GuesserBase
+import numpy as np
+import warnings
+import math
+
+import re
+
+from ..lib import distances
+from . import tables
+
+
+class DefaultGuesser(GuesserBase):
+    """
+    This guesser holds generic methods (not directed to specific contexts) for
+    guessing different topology attribute. It has the same methods that was
+    originally found in Topology.guesser.py. The attributes that can be
+    guessed by this class are:
+    * masses
+    * types
+    * elements
+    * angles
+    * dihedrals
+    * bonds
+    * improper dihedrals
+    * aromaticities
+
+    You can use this guesser either directly through an instance, or through
+    the :meth:`~MDAnalysis.core.universe.Universe.guess_TopologyAttrs` method.
+
+    Examples
+    --------
+    to guess bonds for a universe::
+
+        import MDAnalysis as mda
+        from MDAnalysisTests.datafiles import two_water_gro
+
+        u = mda.Universe(two_water_gro, context='default', to_guess=['bonds'])
+
+    .. versionadded:: 2.8.0
+
+    """
+    context = 'default'
+
+    def __init__(self, universe, **kwargs):
+        super().__init__(universe, **kwargs)
+        self._guesser_methods = {
+            'masses': self.guess_masses,
+            'types': self.guess_types,
+            'elements': self.guess_types,
+            'bonds': self.guess_bonds,
+            'angles': self.guess_angles,
+            'dihedrals': self.guess_dihedrals,
+            'impropers': self.guess_improper_dihedrals,
+            'aromaticities': self.guess_aromaticities,
+        }
+
+    def guess_masses(self, atom_types=None, indices_to_guess=None):
+        """Guess the mass of many atoms based upon their type.
+        For guessing masses through Univese.guess_TopologyAttrs():
+        First it try to guess masses from atom elements, if not available,
+        try to guess masses from types and if not availaible, try to guess
+        types.
+
+        Parameters
+        ----------
+        atom_types
+          Atom types/elements to guess masses from
+        indices_to_guess (optional)
+          Mask array for partially guess masses for certain atoms
+
+        Returns
+        -------
+        atom_masses : np.ndarray dtype float64
+
+        Raises
+        ------
+        :exc:`ValueError` if there is no atom types
+         or elements to guess mass from.
```

Comment:
Here and everywhere else you have this pattern, your explanation string is getting squashed because you don't have a newline, it should be something like this I believe:

```suggestion
        :exc:`ValueError`
            If there are no atom types or elements to guess mass from.
```